base_app/src/app/app-tab-overlay/app-tab-overlay.component.ts

```diff
@@ -392,16 +400,21 @@ export class AppTabOverlayComponent implements AfterViewInit, OnInit {
         console.log('prevStyle["left"]): ' + prevStyle['left'] + ', prevStyle["top"] ' + prevStyle['top']);
         console.log('          ev.pageX: ' + ev.pageX + ', ev.pageY: ' + ev.pageY);
         console.log('          ev.scrollX: ' + ev.scrollX + ', ev.scrollY: ' + ev.scrollY);
-        console.log('           startX:'   + moveLogger['dragStartInfo']['startX'] + ',   startY: ' + moveLogger['dragStartInfo']['startY']);
+        console.log('           startX:'   + moveLogger['dragStartInfo']['startX']
+            + ',   startY: ' + moveLogger['dragStartInfo']['startY']);
         console.log('leftOffsetToMouse:' + moveLogger['dragStartInfo']['leftOffsetToMouse'] +
             ',topOffsetToMouse: ' + moveLogger['dragStartInfo']['topOffsetToMouse']);
 
-        const newHeight: any = e && e.children && e.children[1] && e.children[1].offsetHeight || e.offsetHeight;
+        const newTop: any = Math.round(ev.pageY - anchorCoords.top + moveLogger['dragStartInfo']['topOffsetToMouse']);
+        const newLeft: any = Math.round(ev.pageX - anchorCoords.left + moveLogger['dragStartInfo']['leftOffsetToMouse']);
+
+        let newHeight: any = e && e.children && e.children[1] && e.children[1].offsetHeight || e.offsetHeight;
```

Comment:
you don't need all the ": any" type annotations, can just leave them out. In general 'any' should be used very sparingly since you're effectively telling the type system to ignore whatever you're doing.

Reply:
Will change type to `number`.

Sadly we weren't (I wasn't) disciplined enough to use real types everywhere.
A lot of code was developed prototypically, starting simply from an object and adding fields as necessary, but  the follow up should have been to complete the working using exact types.

We need to scrub out the `any` types from the typescript, but probably won't enter a dedicated issue for it.
Prefer to do it as part of other refactoring / cleanup efforts.